tests/unit/shared/test_vic_time.py

```diff
@@ -75,100 +85,153 @@ def test_fractional_day_from_dmy(dmy_feb_3_noon):
+    for cal in ['julian', 'standard', 'gregorian', 'proleptic_gregorian']:
+        now_jd = JulianDayFromDate(now, calendar=cal)
+        vic_lib.dmy_julian_day(now_jd, calendars[cal], dmy_struct)
+        actual = dmy_to_datetime(dmy_struct)
+        # assert that the difference is less than one second
+        assert abs(now - actual) < datetime.timedelta(seconds=1)
 
 
 @pytest.mark.skipif(nctime_unavailable, reason=nc_reason)
 def test_dmy_no_leap_day():
-    pass
+    dmy_struct = ffi.new("dmy_struct *")
+    now = datetime.datetime.now()
+    now_jd = _NoLeapDayFromDate(now)
+    vic_lib.dmy_no_leap_day(now_jd, dmy_struct)
+    vic_lib.print_dmy(dmy_struct)
+    actual = dmy_to_datetime(dmy_struct)
+    # assert that the difference is less than one second
+    assert abs(now - actual) < datetime.timedelta(seconds=1)
 
 
 @pytest.mark.skipif(nctime_unavailable, reason=nc_reason)
 def test_dmy_all_leap():
-    pass
+    dmy_struct = ffi.new("dmy_struct *")
+    now = datetime.datetime.now()
+    now_jd = _AllLeapFromDate(now)
+    vic_lib.dmy_all_leap(now_jd, dmy_struct)
+    vic_lib.print_dmy(dmy_struct)
+    actual = dmy_to_datetime(dmy_struct)
+    # assert that the difference is less than one second
+    assert abs(now - actual) < datetime.timedelta(seconds=1)
 
 
 @pytest.mark.skipif(nctime_unavailable, reason=nc_reason)
 def test_dmy_all_30_day():
-    pass
+    dmy_struct = ffi.new("dmy_struct *")
+    now = datetime.datetime.now()
+    now_jd = _360DayFromDate(now)
+    vic_lib.dmy_all_30_day(now_jd, dmy_struct)
+    vic_lib.print_dmy(dmy_struct)
+    actual = dmy_to_datetime(dmy_struct)
+    # assert that the difference is less than one second
+    assert abs(now - actual) < datetime.timedelta(seconds=1)
 
 
 @pytest.mark.skipif(nctime_unavailable, reason=nc_reason)
 def test_num2date():
-    pass
+
+    dates = pd.date_range(start="1900-01-01", end="2100-12-31",
+                          freq='MS').to_pydatetime()
+
+    dmy_struct = ffi.new("dmy_struct *")
+    for cal, cal_num in calendars.items():
+        ut = utime(vic_default_units, calendar=cal)
+        for date in dates:
+            num = ut.date2num(date)
+            vic_lib.num2date(ut._jd0, num, 0., cal_num, units['days'],
+                             dmy_struct)
+            actual = dmy_to_datetime(dmy_struct)
+            assert abs(date - actual) < datetime.timedelta(seconds=1)
 
 
 @pytest.mark.skipif(nctime_unavailable, reason=nc_reason)
 def test_make_lastday():
-    lastday = np.zeros(12)
+    # wasn't able to map lastday to a numpy array, don't know why...
+    lastday = ffi.new('unsigned short int []', [0] * 12)
     for year in np.arange(1900, 2100):
-        for cal in ['standard', 'gregorian', 'proleptic_gregorian', 'julian']:
-            vic_lib.make_lastday(calendars[cal], year,
-                                 ffi.addressof(lastday).ctypes.data)
+        for cal in ['standard', 'gregorian', 'proleptic_gregorian']:
```

Comment:
This is just testing whether an individual year is a leap year or not.  For the standard calendars (`['standard', 'gregorian', 'proleptic_gregorian']`), we can use the python calendar module to test against.  The julian calendar isn't available to us via the `calendar` package.  

I think the date2num and num2date tests have sufficiently shown that we are getting the same answers as the `netcdftime` package.